langwatch/src/features/onboarding/components/sections/model-provider/ModelProviderConfigFields.tsx

```diff
@@ -0,0 +1,261 @@
+
+export const ModelProviderConfigFields: React.FC<ModelProviderConfigFieldsProps> = ({
+  modelProviderKey,
+}) => {
+  const meta = getModelProvider(modelProviderKey);
+  const { project } = useOrganizationTeamProject();
+  const projectId = project?.id;
+
+  const backendKey = meta?.backendKey;
+  const { providers, isLoading, refetch } = useModelProvidersSettings({
+    projectId,
+  });
+
+  const provider = useMemo(() => {
+    if (!backendKey) return void 0;
+
+    const existing = providers?.[backendKey as keyof typeof providers];
+    if (existing) return existing;
+
+    return {
+      provider: backendKey,
+      enabled: false,
+      customKeys: null,
+      models: null,
+      embeddingsModels: null,
+      disabledByDefault: true,
+      extraHeaders: [],
+    } as any;
+  }, [backendKey, providers]);
+
+  const [state, actions] = useModelProviderForm({
+    provider,
+    projectId,
+    projectDefaultModel: meta?.defaultModel ?? project?.defaultModel ?? null,
+    onSuccess: () => {
+      void refetch();
+    },
+  });
+
+  const { fields: derivedFields } = useModelProviderFields(backendKey as any);
+  const [openAiValidationError, setOpenAiValidationError] = useState<string>();
+  const [fieldErrors, setFieldErrors] = useState<Record<string, string>>({});
+
+  useEffect(() => {
+    setOpenAiValidationError(void 0);
+    setFieldErrors({});
+  }, [modelProviderKey]);
+
+  const isOpenAiProvider = backendKey === "openai";
+
+  const handleCustomKeyChange = useCallback(
+    (key: string, value: string) => {
+      actions.setCustomKey(key, value);
+    },
+    [actions],
+  );
+
+  const handleFieldErrorClear = useCallback((key: string) => {
+    setFieldErrors((previous) => {
+      if (!previous[key]) {
+        return previous;
+      }
+      const updated = { ...previous };
+      delete updated[key];
+      return updated;
+    });
+  }, []);
+
+  const handleOpenAiValidationClear = useCallback(() => {
+    setOpenAiValidationError(undefined);
+  }, []);
+
+  const validateOpenAi = useCallback(() => {
+    if (!isOpenAiProvider) return true;
+
+    const apiKey = state.customKeys.OPENAI_API_KEY?.trim() ?? "";
+    const baseUrl = state.customKeys.OPENAI_BASE_URL?.trim() ?? "";
+
+    // Both empty
+    if (!apiKey && !baseUrl) {
+      setOpenAiValidationError(
+        "Either API Key or Base URL must be provided",
+      );
+      return false;
+    }
+
+    // Base URL is set to default OpenAI URL, but no API key
+    if (baseUrl === OPENAI_DEFAULT_BASE_URL && !apiKey) {
+      setOpenAiValidationError(
+        "API Key is required when using the default OpenAI base URL",
+      );
+      return false;
+    }
+
+    handleOpenAiValidationClear();
+    return true;
+  }, [
+    isOpenAiProvider,
+    state.customKeys,
+    handleOpenAiValidationClear,
+  ]);
+
+  const handleSaveAndContinue = useCallback(() => {
+    if (!validateOpenAi()) {
+      return;
+    }
```

Comment:
Warning? Communication to the user?

Reply:
That is done inside the `validateOpenAi`. if returns `false` to indicate it's handled it and the caller should exit.